docs/reference/ingest/ingest-node.asciidoc

```diff
@@ -577,6 +577,554 @@ value of `service` to the value of the field `code`:
+POST test/_doc/3?pipeline=drop_guests_network
+{
+  "network.name": "Guest"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+////
+Hidden example assertion:
+[source,js]
+--------------------------------------------------
+GET test/_doc/3
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+// TEST[catch:missing]
+
+[source,js]
+--------------------------------------------------
+{
+  "_index": "test",
+  "_type": "_doc",
+  "_id": "3",
+  "found": false
+}
+--------------------------------------------------
+// TESTRESPONSE
+////
+
+The `?.` operators works well for use in the `if` conditional
+because the {painless}/painless-operators-reference.html#null-safe-operator[null safe operator]
+returns null if the object is null and `==` is null safe (as well as many other
+{painless}/painless-operators.html[painless operators]).
+
+However, calling a method such as `.equalsIgnoreCase` is not null safe
+and can result in a NullPointerException.
+
+Some situations allow for the same functionality but done so in a null safe manner.
+For example: `'Guest'.equalsIgnoreCase(ctx.network?.name)` is null safe because
+`Guest` is always non null, but `ctx.network?.name.equalsIgnoreCase('Guest') is not null safe
+since `ctx.network?.name` can return null.
+
+Some situations require an explicit null check. In the following example there
+is not null safe alternative, so an explict null check is needed.
+
+[source,js]
+--------------------------------------------------
+{
+  "drop": {
+    "if": "ctx.network?.name != null && ctx.network.name.contains('Guest')"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+[[ingest-conditional-complex]]
+=== Complex Conditionals
+The `if` condition can be more then a simple equality check.
+The full power of the <<modules-scripting-painless, Painless Scripting Language>> ability to declare `tag` and `on_failure` fields. These fields are optional
+running in the {painless}/painless-ingest-processor-context.html#null-safe-operator[ingest processor context]
+can be used to determine if the condition should evaluate to `true` or `false`.
+
+IMPORTANT: Values in an `if` condition are read-only and may not change any
+values of `ctx`.
+
+A more complex `if` condition that drops the document (e.g. not index it)
+unless it has a multi-valued tag field with at least one value that contains the characters
+`prod` (case insensitive).
+
+[source,js]
+--------------------------------------------------
+PUT _ingest/pipeline/not_prod_dropper
+{
+  "processors": [
+    {
+      "drop": {
+        "if": "Collection tags = ctx.tags;if(tags != null){for (String tag : tags) {if (tag.toLowerCase().contains('prod')) { return false;}}} return true;"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+// CONSOLE
+
+The conditional needs to be all on one line since JSON does not
```

Comment:
This seems like it should be explained elsewhere. This is a very general fact about scripts.

Reply:
yeah, i really wanted the `if` conditional (script) to be readable on the page which requires the triple quote, but only wanted to provide a CONSOLE example that is valid JSON. Not sure how to achieve readability and valid JSON without having to repeat this fact.